althea_types/src/wifi_info.rs

```diff
@@ -0,0 +1,471 @@
+/// This struct contains information for wifi survey data
+#[derive(Default, Debug, Clone, Serialize, Deserialize)]
+pub struct WifiSurveyData {
+    /// Frequency in MHz
+    pub frequency_mhz: u16,
+    /// Noise in dBm
+    pub noise_dbm: i32,
+    /// Time in ms
+    pub channel_active_time: u64,
+    pub channel_busy_time: u64,
+    pub channel_receive_time: u64,
+    pub channel_transmit_time: u64,
+}
+
+/// This struct contains information for wifi station data
+#[derive(Default, Debug, Clone, Serialize, Deserialize)]
+pub struct WifiStationData {
+    pub station: String,
+    pub inactive_time_ms: u64,
+    pub rx_bytes: u64,
+    pub rx_packets: u64,
+    pub tx_bytes: u64,
+    pub tx_packets: u64,
+    pub tx_retries: u16,
+    pub tx_failed: u16,
+    pub rx_drop_misc: u16,
+    pub signal_dbm: String,
```

Comment:
can't we used a signed integer here?

Reply:
```
root@OpenWrt:~# iw wlan1 station dump
Station <mac removed> (on wlan1)
	inactive time:	10 ms
	rx bytes:	141627376
	rx packets:	329289
	tx bytes:	1106901
	tx packets:	6172
	tx retries:	0
	tx failed:	0
	rx drop misc:	322643
	signal:  	-73 dBm
	signal avg:	-73 dBm
	tx bitrate:	6.0 MBit/s 40MHz
	rx bitrate:	54.0 MBit/s
	rx duration:	0 us
	authorized:	yes
	authenticated:	yes
	associated:	yes
	preamble:	short
	WMM/WME:	yes
	MFP:		no
	TDLS peer:	no
	DTIM period:	2
	beacon interval:100
	short preamble:	yes
	short slot time:yes
	connected time:	26651 seconds
```

Just grabbed this sample from an older router, it looks to me like some of these fields (airtime_weight, associated_at, current_time) may need to be options.

Reply:
signal:      -59 [-68, -59, -100, -100] dBm
    signal avg:    -58 [-67, -58, -100, -100] dBm